### PR TITLE
ref: Rename SentryInAppLogic

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Rename SentryFrameInAppLogic to SentryInAppLogic, because the class doesn't
check if a frame is in app or not, but if frameworks are in app or not.

#skip-changelog

## :bulb: Motivation and Context

I came across this when investigating https://github.com/getsentry/sentry-cocoa/issues/1229 and preparing a fix for it.

## :green_heart: How did you test it?
CI

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryOptions.m [hunk 1]
```diff
@@ -48,7 +48,7 @@ - (instancetype)init
         _defaultTracesSampleRate = nil;
         self.tracesSampleRate = _defaultTracesSampleRate;
 
-        // Use the name of the bundle’s executable file as inAppInclude, so SentryFrameInAppLogic
+        // Use the name of the bundle’s executable file as inAppInclude, so SentryInAppLogic
         // marks frames coming from there as inApp. With this approach, the SDK marks public
```
Comment (bitsandfoxes) [inline Sources/Sentry/SentryOptions.m:52]: So no more frames get marked?

Comment (philipphofmann) [reply Sources/Sentry/SentryOptions.m:52]: You still can use the class to check if a frame is in-app or not, but you don't do that by passing in the frame but instead the image name. See 

> // Use the name of the bundle’s executable file as inAppInclude, so SentryInAppLogic
> // marks frames coming from there as inApp. With this approach, the SDK marks public

**Review by bitsandfoxes — APPROVED** [review]
LGTM!

File: Sources/Sentry/SentryOptions.m (hunk 1 above)
Comment (bitsandfoxes) [reply Sources/Sentry/SentryOptions.m:52]: Oh, I see. That makes sense.

**Pull request merged**